pkg/plugins/discovery.go

```diff
@@ -76,3 +77,31 @@ func readPluginInfo(path string, fi os.FileInfo) (*Plugin, error) {
 		Addr: addr,
 	}, nil
 }
+
+func readPluginSocketInfo(path string, fi os.FileInfo) (*Plugin, error) {
+	name := strings.Split(fi.Name(), ".")[0]
+
+	if fi.Mode()&os.ModeSocket == 0 {
```

Comment:
> Would love a comment with a link to what this mask is matching

I'd rather not add comments in our code about how Go's standard library works:

http://golang.org/pkg/os/#FileMode